addons/sale_stock/sale_stock.py

```diff
@@ -281,6 +281,25 @@ def product_packaging_change(self, cr, uid, ids, pricelist, product, qty=0, uom=
 
         return {'value': result, 'warning': warning}
 
+    def _check_routing(self, cr, uid, product, warehouse, context=None):
+        isMto = False
+        if warehouse:
+            for product_route in product.route_ids:
+                if warehouse.mto_pull_id and warehouse.mto_pull_id.route_id and warehouse.mto_pull_id.route_id.id == product_route.id:
+                    isMto = True
+                    break
+        else:
```

Comment:
Speaking considering SO it is required but speaking with ref of product_id_change_with_wh(...) it is an optional parameter, so else needs to be there.